Console.Flashcards.mefdev/FlashcardApp.Core/Repositories/FlashcardRepository.cs

```diff
@@ -0,0 +1,164 @@
+                return flashcard;
+            }, splitOn: "stackId");
+
+            return flashcard;
+        }
+        catch (Exception ex)
+        {
+            throw new Exception($"{ex.Message}");
+        }
+
+    }
+
+    public async Task<Flashcard> GetFlashcard(int id)
+    {
+        try
+        {
+            using var db = _dbContext.CreateConnection();
+            string query = "SELECT * FROM FlashCards fc LEFT JOIN Stacks s on fc.StackId = s.stackId WHERE fc.flashcardId=@id";
+            var flashCard = await db.QueryAsync<Flashcard, Stack, Flashcard>(query, (flashcard, stack) =>
+            {
+                flashcard.Stack = stack;
+                return flashcard;
+            },
+            new { id = id },
+            splitOn: "stackId");
+
+            var flashcard = flashCard.FirstOrDefault();
+            if (flashcard == null)
+            {
+                Console.Error.WriteLine("Notice: The flashcard cannot be found");
+                return null;
+            }
+            return flashcard;
+        }
+        catch (Exception ex)
+        {
+            throw new Exception($"{ex.Message}");
+        }
+
+    }
+
+    public async Task<Flashcard> GetFlashcardByQuestion(string question)
+    {
+        try
+        {
+            using var db = _dbContext.CreateConnection();
+            string query = "SELECT * FROM FlashCards f LEFT JOIN Stacks s on f.StackId = s.stackId WHERE f.Question=@question";
+            var flashCard = await db.QueryAsync<Flashcard, Stack, Flashcard>(query, (flashcard, stack) =>
+            {
+                flashcard.Stack = stack;
+                return flashcard;
+            },
+            new { question = question },
+            splitOn: "stackId");
+
+            var flashcard = flashCard.FirstOrDefault();
+            if (flashcard == null)
+            {
+                Console.Error.WriteLine("Notice: The flashcard cannot be found");
+                return null;
+            }
+            return flashcard;
+        }
+        catch (Exception ex)
+        {
+            throw new Exception($"{ex.Message}");
+        }
+    }
+
+    public async Task<IEnumerable<Flashcard>> GetFlashcardsByStackname(string name)
+    {
+        try
+        {
+            using var db = _dbContext.CreateConnection();
+            string query = "SELECT * FROM FlashCards f LEFT JOIN Stacks s on f.StackId = s.stackId WHERE s.Name = @name";
+            var flashcard = await db.QueryAsync<Flashcard, Stack, Flashcard>(query, (flashcard, stack) =>
+            {
+                flashcard.Stack = stack;
+                return flashcard;
+            }, new {name = name},
+            splitOn: "stackId");
+            return flashcard;
+        }
+        catch (Exception ex)
+        {
+            throw new Exception($"{ex.Message}");
+        }
+    }
+
+    public Task UpdateFlashcard(Flashcard Flashcard)
+    {
+        throw new NotImplementedException();
```

Comment:
Is this intended?

Reply:
Yes, it is intended because I did not need to update the flashcard.